core/src/main/java/com/netflix/conductor/domain/WorkflowStatusDO.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.domain;
+
+import com.netflix.conductor.common.run.Workflow.WorkflowStatus;
+
+public enum WorkflowStatusDO {
```

Comment:
I think we should abstract the status out into its own object model, but do we need to separate Status into DO, DTO objects? What used internally for the states machines should be the same for what exposed to users, right? My concern is about keep it consistently and considerate each other if separated. Same goes for TaskStatus

Reply:
nested the status object inside the model